bundles/org.openhab.binding.tado/src/main/java/org/openhab/binding/tado/internal/api/TadoApiTypeUtils.java

```diff
@@ -39,9 +42,11 @@
  *
  * @author Dennis Frommknecht - Initial contribution
  */
+@NonNullByDefault
 public class TadoApiTypeUtils {
+
     public static OverlayTerminationCondition getTerminationCondition(OverlayTerminationConditionType type,
-            Integer timerDurationInSeconds) {
+            @Nullable Integer timerDurationInSeconds) {
```

Comment:
Do we expect `timerDurationInSeconds` being null while type is `TIMER`? If not, you could also throw `IllegalArgumentException` in this case and avoid `@Nullable` in `timerTermination`.